.github/workflows/changelog-checker.yml

```diff
@@ -14,11 +14,22 @@ jobs:
         uses: actions/checkout@v4
         with:
           path: repo
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

Comment:
This should use a specific hash rather than just `v4`. I think it could be worth also rebasing or merging main, which already has actions/checkout pinned.